master/internal/user/postgres_users.go

```diff
@@ -141,38 +118,33 @@ func addUserExec(ctx context.Context, user *model.User) (model.UserID, error) {
 		UserID:  user.ID,
 		GroupID: personalGroup.ID,
 	}
-	if _, err = tx.NewInsert().Model(&groupMembership).Exec(ctx); err != nil {
+	if _, err = idb.NewInsert().Model(&groupMembership).Exec(ctx); err != nil {
 		return 0, errors.Wrap(err, "error adding user to personal group")
 	}
-
-	if err = tx.Commit(); err != nil {
-		return 0, errors.Wrap(err, "error committing changes in addUserExec")
-	}
-
 	return user.ID, nil
 }
 
-// AddUser creates a new user.
-func AddUser(
+// Add creates a new user.
+func Add(
 	ctx context.Context,
 	user *model.User,
 	ug *model.AgentUserGroup,
 ) (model.UserID, error) {
-	userID, err := addUserExec(ctx, user)
+	userID, err := addExec(ctx, db.Bun(), user)
```

Comment:
Use `db.Bun().RunInTx` and pass these the tx (or make a tx manually), so that the multiple inserts are atomic.